experiment/coverage_utils.py

```diff
@@ -0,0 +1,206 @@
+    generator = CoverageReporter(fuzzer, benchmark, experiments, report_dir,
+                                 logger)
+    # Gets and merges all the profdata files.
+    generator.fetch_profdata_file()
+    generator.merge_profdata_files()
+    # Generates the reports using llvm-cov.
+    generator.generate_cov_report()
+
+    logger.info('Finished generating coverage report for '
+                'benchmark:{benchmark} fuzzer:{fuzzer}.'.format(
+                    benchmark=benchmark, fuzzer=fuzzer))
+
+
+def set_up_coverage_files(experiment_names, report_dir, benchmarks):
+    """Sets up coverage files for all benchmarks."""
+    with multiprocessing.Pool() as pool:
+        set_up_coverage_file_args = [(experiment_names, report_dir, benchmark)
+                                     for benchmark in benchmarks]
+        pool.starmap(set_up_coverage_file, set_up_coverage_file_args)
+        pool.close()
+        pool.join()
+
+
+def set_up_coverage_file(experiment_names, report_dir, benchmark):
+    """Sets up coverage files for |benchmark|."""
+    logs.initialize()
+    logger.info('Started setting up coverage file for'
+                'benchmark: {benchmark}'.format(benchmark=benchmark))
+    for experiment in experiment_names:
+        archive_filestore_path = get_benchmark_archive(experiment, benchmark)
+        archive_exist = filestore_utils.ls(archive_filestore_path,
+                                           must_exist=False).retcode == 0
+        if archive_exist:
+            benchmark_report_dir = get_benchmark_report_dir(
+                benchmark, report_dir)
+            filesystem.create_directory(benchmark_report_dir)
+            filestore_utils.cp(archive_filestore_path,
+                               str(benchmark_report_dir))
+            print(benchmark_report_dir)
+            archive_name = 'coverage-build-%s.tar.gz' % benchmark
+            archive_path = os.path.join(benchmark_report_dir, archive_name)
+            tar = tarfile.open(archive_path, 'r:gz')
+            tar.extractall(benchmark_report_dir)
+            os.remove(archive_path)
+            break
+    logger.info('Finished setting up coverage file for'
+                'benchmark: {benchmark}'.format(benchmark=benchmark))
+
+
+def get_benchmark_archive(experiment_name, benchmark):
+    """Returns the path of the coverage archive in gcs bucket
+    for |benchmark|."""
+    experiment_filestore_dir = get_experiment_filestore_path(experiment_name)
+    archive_name = 'coverage-build-%s.tar.gz' % benchmark
+    return posixpath.join(experiment_filestore_dir, 'coverage-binaries',
+                          archive_name)
+
+
+def get_experiment_filestore_path(experiment_name):
+    """Returns the path of the storage folder for |experiment_name|."""
+    if 'EXPERIMENT_FILESTORE' in os.environ:
+        experiment_filestore = os.environ['EXPERIMENT_FILESTORE']
+    else:
+        experiment_filestore = 'gs://fuzzbench-data'
```

Comment:
NO hardcode, just add helper in experiment_utils to get this and assume it is always there.

Reply:
Does it mean we should expect the user to set this environment variable when running `generate_report` locally? I thought that may be a burden for the user and most of them would expect to get the experiment-data from `gs://fuzzbench-data`; so I coded this way.